.github/shared/src/sdk-types.js

```diff
@@ -1,7 +1,38 @@
-/* v8 ignore start */
+import * as z from "zod";
 
+const SdkNameSchema = z.enum([
+  "azure-sdk-for-go",
+  "azure-sdk-for-java",
+  "azure-sdk-for-js",
+  "azure-sdk-for-net",
+  "azure-sdk-for-python",
+]);
 /**
- * @typedef {'azure-sdk-for-go' | 'azure-sdk-for-java' | 'azure-sdk-for-js' | 'azure-sdk-for-net' | 'azure-sdk-for-python'} SdkName
+ * @typedef {import("zod").infer<typeof SdkNameSchema>} SdkName
+ */
+
+/*
+ * Data for the API view request.
+ */
+const APIViewRequestDataSchema = z.object({ packageName: z.string(), filePath: z.string() });
+/**
+ * @typedef {import("zod").infer<typeof APIViewRequestDataSchema>} APIViewRequestData
+ */
+
+/**
+ * Represents the result of the spec-gen-sdk generation process.
+ */
+export const SpecGenSdkArtifactInfoSchema = z.object({
+  language: SdkNameSchema,
+  result: z.string(),
+  headSha: z.string(),
+  prNumber: z.string().optional(),
+  labelAction: z.boolean().optional(),
+  isSpecGenSdkCheckRequired: z.boolean(),
+  apiViewRequestData: z.array(APIViewRequestDataSchema),
+});
+/**
+ * @typedef {import("zod").infer<typeof SpecGenSdkArtifactInfoSchema>} SpecGenSdkArtifactInfo
```

Comment:
type `SpecGenSdkArtifactInfo` should be the same as the copy removed from `eng/tools/spec-gen-sdk-runner/src/types.ts`